File: count_ways_to_build_good_strings.cpp
class Solution {
public:
    int countGoodStrings(int low, int high, int zero, int one) {
        vector<int> dp(high+1);
        dp[0] = 1;
        int mod = 1000000007;

        for (int i = 1; i <= high; i++) {
            if (i - zero >= 0) {
                dp[i] += dp[i-zero];
            }
            
            if (i - one >= 0) {
                dp[i] += dp[i-one];
            }
            
            dp[i] %= mod;
        }
        
        int res = 0;
        for (int i = low; i <= high; i++) {
            res += dp[i];
            res %= mod;
        }
        
        return res;        
    }
};

File: count_ways_to_build_good_strings.py
class Solution:
    def countGoodStrings(self, low: int, high: int, zero: int, one: int) -> int:
        dp = [1] + [0] * high
        mod = 10 ** 9 + 7

        for i in range(1, high + 1):
            if i - zero >= 0:
                dp[i] += dp[i-zero]
            
            if i - one >= 0:
                dp[i] += dp[i-one]
            
            dp[i] %= mod
        
        return sum(dp[low:high+1]) % mod
